src/commands/group_by.rs

```diff
@@ -0,0 +1,89 @@
+use crate::commands::WholeStreamCommand;
+use crate::data::TaggedDictBuilder;
+use crate::errors::ShellError;
+use crate::prelude::*;
+
+pub struct GroupBy;
+
+#[derive(Deserialize)]
+pub struct GroupByArgs {
+    column_name: Tagged<String>,
+}
+
+impl WholeStreamCommand for GroupBy {
+    fn name(&self) -> &str {
+        "group-by"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("group-by").required("column_name", SyntaxShape::String)
+    }
+
+    fn usage(&self) -> &str {
+        "Creates a new table with the data from the table rows grouped by the column given."
+    }
+
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        args.process(registry, group_by)?.run()
+    }
+}
+
+fn group_by(
+    GroupByArgs { column_name }: GroupByArgs,
+    RunnableContext { input, name, .. }: RunnableContext,
+) -> Result<OutputStream, ShellError> {
+    let stream = async_stream! {
+        let values: Vec<Tagged<Value>> = input.values.collect().await;
+        let mut groups = indexmap::IndexMap::new();
+
+        for row in values {
+            let key = row.get_data_by_key(&column_name.item);
+
+            if key.is_none() {
+
+                let possibilities = row.data_descriptors();
+
+                let mut possible_matches: Vec<_> = possibilities
+                    .iter()
+                    .map(|x| (natural::distance::levenshtein_distance(x, &column_name.item), x))
+                    .collect();
+
+                possible_matches.sort();
+
+                let err = {
+                    if possible_matches.len() > 0 {
+                        ShellError::labeled_error(
+                            "Unknown column",
+                            format!("did you mean '{}'?", possible_matches[0].1),
+                            &column_name.tag,)
+                    } else {
+                        ShellError::labeled_error(
+                            "Unknown column",
+                            "row does not contain this column",
+                            &column_name.tag,
+                        )
+                    }
+                };
+
+                yield Err(err)
+            }
+
+            let mut group = groups.entry(key.unwrap().as_string()?).or_insert(vec![]);
```

Comment:
You may want to put these lines in an `else`.  Withtout it, if there's an error, we'll send the error, but then still unwrap and potentially panic.